File: app/src/main/java/alireza/nezami/dependency_injection_without_dagger/java/approach1/data/remote/NetworkService.java
package alireza.nezami.dependency_injection_without_dagger.java.approach1.data.remote;

import android.content.Context;

/**
 * Dummy class to simulate the actual NetworkService using Retrofit or OkHttp etc
 */
public class NetworkService {

    private Context context;
    private String apiKey;

    public NetworkService(Context context) {
        this.context = context;
        // do the initialisation here
        apiKey = "SOME_API_KEY";
    }

    public String getDummyData() {
        return "NETWORK_DUMMY_DATA";
    }
}

File: app/src/main/java/alireza/nezami/dependency_injection_without_dagger/kotlin/approach1/data/remote/NetworkService.kt
package alireza.nezami.dependency_injection_without_dagger.kotlin.approach1.data.remote

import android.content.Context

/**
 * Dummy class to simulate the actual NetworkService using Retrofit or OkHttp etc
 */
class NetworkService
    (
    private val context: Context?,
    private val apiKey: String = "SOME_API_KEY"
) {
    val dummyData: String
        get() = "NETWORK_DUMMY_DATA"
}